File: apps/commerce-api/src/main/java/com/loopers/domain/product/ProductOption.java
package com.loopers.domain.product;

import java.math.BigDecimal;

import com.loopers.domain.BaseEntity;

import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "product_option")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class ProductOption extends BaseEntity {
    private String optionType;
    private String optionValue;
    private BigDecimal additionalPrice;

    public ProductOption(String optionType, String optionValue, BigDecimal additionalPrice) {
        this.optionType = optionType;
        this.optionValue = optionValue;
        this.additionalPrice = additionalPrice;
    }
}

File: apps/commerce-streamer/src/main/kotlin/com/loopers/domain/product/ProductOption.kt
package com.loopers.domain.product

import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table(name = "product_option")
data class ProductOption(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long?,
    val optionType: String,
    val optionValue: String,
    val additionalPrice: BigDecimal,
    @OneToOne(mappedBy = "productOption")
    val inventory: Inventory
) {
    fun deduct(quantity: Int) {
        inventory.deduct(quantity)
    }
}
